app-react/src/main/kotlin/com/ustadmobile/view/HolidayCalendarDetailScreen.kt

```diff
@@ -0,0 +1,87 @@
+package com.ustadmobile.view
+
+import com.ustadmobile.core.generated.locale.MessageID
+import com.ustadmobile.core.hooks.useStringsXml
+import com.ustadmobile.core.impl.locale.StringsXml
+import com.ustadmobile.core.viewmodel.HolidayCalendarDetailUIState
+import com.ustadmobile.lib.db.entities.Holiday
+import com.ustadmobile.lib.db.entities.HolidayCalendar
+import com.ustadmobile.mui.components.UstadDetailField
+import mui.material.*
+import mui.system.responsive
+import react.FC
+import react.Props
+import react.useMemo
+import kotlin.js.Date
+
+external interface HolidayCalendarDetailProps: Props{
+    var uiState: HolidayCalendarDetailUIState
+    var onItemClick: (Holiday) -> Unit
+}
+
+val HolidayCalendarDetailComponent2 = FC<HolidayCalendarDetailProps> { props ->
+
+    val strings: StringsXml = useStringsXml()
+
+    Container{
+        Stack{
+            spacing = responsive(2)
+
+            UstadDetailField{
+                labelText = strings[MessageID.name]
+                valueText = props.uiState.holidayCalendar?.umCalendarName ?: ""
+            }
+
+            List{
+                props.uiState.holidayList?.forEach { item ->
+                    ListItem {
+
+                        disablePadding = true
+
+                        val holidayStart = useMemo(item.holStartTime) {
+                            Date(item.holStartTime).toLocaleDateString()
+                        }
+                        val holidayEnd = useMemo(item.holEndTime) {
+                            Date(item.holEndTime).toLocaleDateString()
+                        }
+
+                        ListItemButton {
+                            onClick = {
+                                props.onItemClick(item)
+                            }
+                            ListItemText {
+                                + (item.holName ?: "")
```

Comment:
You need to use the primary and secondary properties for ListItemText instead of putting the text in directly. ListItemSecondaryAction will appear in the top right corner. That is for secondary action buttons (e.g. download buttons, option menus etc) as per the MaterialDesign guidelines ( https://m2.material.io/components/lists#anatomy ) . 

E.g. it should look like this:

```
ListItemText {
    primary = ReactNode(item.holName ?: "")
    secondary = ReactNode("$holidayStart - $holidayEnd")
}
```